controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -77,6 +80,12 @@ type KubeadmControlPlaneReconciler struct {
 
 	managementCluster         internal.ManagementCluster
 	managementClusterUncached internal.ManagementCluster
+
+	// disableInPlacePropagation should only be used for tests. This is used to skip
+	// some parts of the controller that need SSA as the current test setup does not
+	// support SSA. This flag should be dropped after all affected tests are migrated
+	// to envtest.
+	disableInPlacePropagation bool
```

Comment:
Added a follow-up item in the umbrella issue to drop this and migrate the affected tests to envtest.

Reply:
👍